cmd2/parsing.py

```diff
@@ -270,31 +270,42 @@ class StatementParser:
     Shortcuts is a list of tuples with each tuple containing the shortcut and
     the expansion.
     """
-    def __init__(
-            self,
-            allow_redirection: bool = True,
-            terminators: List[str] = None,
-            multiline_commands: List[str] = None,
-            aliases: Dict[str, str] = None,
-            shortcuts: List[Tuple[str, str]] = None,
-    ):
+    def __init__(self,
+                 allow_redirection: bool = True,
+                 terminators: Optional[Iterable[str]] = None,
+                 multiline_commands: Optional[Iterable[str]] = None,
+                 aliases: Optional[Dict[str, str]] = None,
+                 shortcuts: Optional[Iterable[Tuple[str, str]]] = None) -> None:
+        """Initialize an instance of StatementParser.
+
+        The following will get converted to an immutable tuple before storing internally:
+        * terminators
+        * multiline commands
+        * shortcuts
+
+        :param allow_redirection: (optional) should redirection and pipes be allowed?
+        :param terminators: (optional) iterable containing strings which should terminate multiline commands
+        :param multiline_commands: (optional) iterable containing the names of commands that accept multiline input
+        :param aliases: (optional) dictionary contaiing aliases
+        :param shortcuts (optional) an iterable of tuples with each tuple containing the shortcut and the expansion
+        """
         self.allow_redirection = allow_redirection
         if terminators is None:
-            self.terminators = [constants.MULTILINE_TERMINATOR]
+            self.terminators = (constants.MULTILINE_TERMINATOR,)
         else:
-            self.terminators = terminators
+            self.terminators = tuple(terminators)
         if multiline_commands is None:
-            self.multiline_commands = []
+            self.multiline_commands = ()
         else:
-            self.multiline_commands = multiline_commands
+            self.multiline_commands = tuple(multiline_commands)
         if aliases is None:
```

Comment:
Can aliases ever be none?

Reply:
No, but I did it this way for symmetry with the other arguments passed in to ``StatementParser``.  

If you think it is cleaner, I could remove the default values for the following arguments to ``StatementParser.__init__()`` since they should never be ``None`` when coming from ``cmd2.Cmd``:
* ``aliases``
* ``allow_redirection`` 
* ``shortcuts``

Reply:
With the modifications I made, ``aliases`` can be (and are typically) ``None``